pkg/webhook/resourcedistribution/validating/utils.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2021 The Kruise Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validating
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+)
+
+const (
+	DefaultNamespace                     = "default"
+	ResourceHashCodeAnnotation           = "kruise.io/resourcedistribution.resource.hashcode"
+	SourceResourceDistributionOfResource = "kruise.io/resourcedistribution.resource.from"
+)
+
+var (
+	// SupportGVKList is a list that contains all support resource group, version, and kind
+	// Support CustomResourceDefinition
+	/* ADD NEW RESOURCE TYPE HERE*/
+	supportGVKList = []schema.GroupVersionKind{
+		{Group: "", Version: "v1", Kind: "Secret"},
+		{Group: "", Version: "v1", Kind: "ConfigMap"},
+	}
+
+	// Resources will never be distributed to namespaces in this list
+	/* ADD NEW FORBIDDEN NAMESPACE HERE*/
+	forbiddenNamespaces = []string{
+		"kube-system",
+		"kube-public",
+	}
+)
+
+// SupportResourceGVK check whether object is supported by ResourceDistribution
+func SupportResourceGVK(object runtime.Object) bool {
```

Comment:
is it necessary to publish all functions in these packages ?

Reply:
Some function, such as GetTargetNamespaces, is public because they are reused in controller.
Some such as SupportResourceGVK is public because I thought that other CRD may use ResourceDistribution to distribute some kinds of resource, so I published these functions. Maybe I was wrong. I will modify them as private.